Source/ASCollectionView.mm

```diff
@@ -879,9 +856,21 @@ Performing nested batch updates with super (e.g. resizing a cell node & updating
 - (void)_superPerformBatchUpdates:(void(^)())updates completion:(void(^)(BOOL finished))completion
 {
   ASDisplayNodeAssertMainThread();
-  
+
   _superBatchUpdateCount++;
-  [super performBatchUpdates:updates completion:completion];
+
+  if (ASCellLayoutModeIncludes(ASCellLayoutModeAlwaysReloadData)) {
+    if (updates) {
+      updates();
+    }
+    [super reloadData];
+    if (completion) {
+      completion(YES);
+    }
+  } else {
+    [super performBatchUpdates:updates completion:completion];
+  }
```

Comment:
Why have this fork in two places i.e. here and also below at 1750 and 2214?

Reply:
In the two places we have a different logic when to perform a batch update or reload data. In `rangeController:updateWithChangeSet:updates`: besides checking for the `ASCellLayoutModeAlwaysReloadData` flag we also checking some more conditions to decide if we should reload data or batch update.

That said I removed the "automatic" behavior within `_superPerformBatchUpdates:completion:` and we are now explicitly calling if else in places where we would like to differentiate between the calls. It was not really obvious from a caller perspective that this automatic behavior exists in the first place.